CDCgov/prime-reportstream
# 15051,15157: download report and file name fix

This PR fixes two issues from the updates to how the filename is stored.

Test Steps:
Download issue
1. With any report id generated from the translate, batch or send step call
```
curl --location 'http://localhost:7071/api/history/report/{REPORT_ID}' \
--header 'organization: ignore' \
--header 'Authorization: Bearer dummy'
```
on master, you'll get a 404 and on this branch you'll get the contents

File format
1. Setup a receiver to use one of the custom formats in `file-name-templates`
2. Send a report that will go to that receiver
3. Verify that the file name in the SFTP bucket matches the configured format

## Changes
- Updates getReportId to form the external name if one does not exist on the report
- Updates Report.formExternalFileName to pass in the nameFormat and the translation config

## Checklist

### Testing

### Process

## Linked Issues
- #15051 
- #15157 

## To Be Done


## Specific Security-related subjects a reviewer should pay specific attention to
